### PR TITLE
update ScrollMetricsNotification

### PR DESCRIPTION
**Pull request by xu-baolin**
1, notify the first scroll metrics initialize change(null -> non-null)
2,remove the first empty scroll notification request

CC @Piinks @chunhtai

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1224,25 +1224,16 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    _maybeTriggerScrollbar();
+    assert(_debugScheduleCheckHasValidScrollPosition());
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1227]: Won't this result in the scrollbar not being triggered in release builds? Asserts are disabled in release.

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 2]
```diff
@@ -1224,25 +1224,16 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
...
+      return true;
     WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
-      final ScrollController? scrollController = widget.controller ?? PrimaryScrollController.of(context);
-      if (showScrollbar) {
-        _fadeoutTimer?.cancel();
-        // Wait one frame and cause an empty scroll event.  This allows the
-        // thumb to show immediately when isAlwaysShown is true. A scroll
-        // event is required in order to paint the thumb.
-        assert(_debugCheckHasValidScrollPosition());
-        scrollController!.position.didUpdateScrollPositionBy(0);
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1243]: Removing this hack is nice! 👍

**Review by Piinks — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 3]
```diff
@@ -1365,7 +1356,8 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     super.didUpdateWidget(oldWidget);
     if (widget.isAlwaysShown != oldWidget.isAlwaysShown) {
       if (widget.isAlwaysShown == true) {
-        _maybeTriggerScrollbar();
+        assert(_debugScheduleCheckHasValidScrollPosition());
+        _fadeoutTimer?.cancel();
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1360]: Oh I see, the rest was factored out of the assert. This is a really nice clean up.

**Review by Piinks — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
Rebasing after https://github.com/flutter/flutter/pull/87518 should fix the failing check. See  https://github.com/flutter/flutter/issues/87517 for more info

File: packages/flutter/lib/src/widgets/scrollbar.dart (hunk 2 above)
Comment (xu-baolin) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1243]: 😄

**Pull request merged**